src/Executor/Setup.cs

```diff
@@ -0,0 +1,290 @@
+                    {
+                        commandValues[optSetting.Key] = value;
+                    }
+                    else
+                    {
+                        return false;
+                    }
+                }
+            }
+            return true;
+        }
+
+        private bool ValidExtraArgumentsForCommand(string extraArguments, Dictionary<string, string> commandValues)
+        {
+            int colonPosition;
+            int equalPosition;
+            string tempParam;
+
+            string[] extraA = extraArguments.Split(' ');
+            foreach(string param in extraA)
+            {
+                colonPosition = 0;
+                equalPosition = param.Length;
+                tempParam = string.Empty;
+
+                colonPosition = param.IndexOf(":");
+                equalPosition = param.IndexOf("=");
+                if(colonPosition!=0)
+                {
+                    if(equalPosition == -1)
+                    {
+                        tempParam = param.Substring(colonPosition + 1, (param.Length - colonPosition - 1));
+                    }
+                    else
+                    {
+                        tempParam = param.Substring(colonPosition + 1, (equalPosition - colonPosition - 1));
+                    }
+
+                    if(commandValues.ContainsKey(tempParam) && !string.IsNullOrEmpty(commandValues[tempParam]))
+                    {
+                        Console.WriteLine("Error: The value for setting {0} can't be overwriten.", tempParam);
+                        return false;
+                    }
+                }
+            }
+            return true;
+        }
+
+        public string GetTool(Command commandToExecute, string os, string configPath)
+        {
+            if (Tools.ContainsKey(commandToExecute.ToolName))
+            {
+                if(commandToExecute.ToolName.Equals("msbuild"))
+                {
+                    return Path.GetFullPath(Path.Combine(configPath, os.Equals("windows") ? Tools[commandToExecute.ToolName].Run["windows"] : Tools[commandToExecute.ToolName].Run["unix"]));
+                }
+                else if (commandToExecute.ToolName.Equals("console"))
+                {
+                    string extension = os.Equals("windows") ? Tools[commandToExecute.ToolName].Run["windows"] : Tools[commandToExecute.ToolName].Run["unix"];
+                    return Path.GetFullPath(Path.Combine(configPath, string.Format("{0}.{1}", commandToExecute.LockedSettings["Project"],extension)));
+                }
+            }
+            return string.Empty;
+        }
+            
+        public string FormatSetting(string option, string value, string type, string toolName)
+        {
+            string commandOption = null;
+            if (type.Equals("passThrough"))
+            {
+                commandOption = string.Format(" {0}", toolName.Equals("console") ? "" : value);
```

Comment:
So the console command ignores any "passThrough" arguments? That's unexpected (at least to me)